models/qwen3/14b/decode_fwd.py

```
@@ -1769,22 +1799,26 @@ def _grow_slot_mapping(seq_lens: torch.Tensor) -> torch.Tensor:
+        ref_out = torch.zeros(BATCH_PAD, HIDDEN, dtype=torch.bfloat16)
         decode_fwd_layers(hs, *stacked[:len(INPUT_NAMES) - 1], ref_out, config=run_cfg)
         hn = ref_out.float()
         inv = torch.rsqrt(hn.pow(2).mean(-1, keepdim=True) + EPS)
         ref_normed = (hn * inv) * final_norm_w.float()
-        ref_logits = ref_normed @ lm_head_w.float().t()  # [BATCH, VOCAB]
-        a = logits.cpu(); e = ref_logits.cpu()
+        ref_logits = ref_normed @ lm_head_w.float().t()  # [BATCH_PAD, VOCAB]
+        # Only rows [0, UB) carry real sequences. The reference shares the UB-row
+        # seq_lens, so decode_fwd_layers computes the same UB rows; rows past UB
+        # are padding on both sides and are deliberately not compared.
+        a = logits.cpu()[:UB]; e = ref_logits.cpu()[:UB]
```

Ruff (0.15.21)

[error] Multiple statements on one line (semicolon)

(E702)